mlir/include/mlir/Dialect/LLVMIR/LLVMAttrDefs.td

```diff
@@ -350,8 +350,21 @@ def LLVM_DICompileUnitAttr : LLVM_Attr<"DICompileUnit", "di_compile_unit",
     "DIFileAttr":$file,
     OptionalParameter<"StringAttr">:$producer,
     "bool":$isOptimized,
-    "DIEmissionKind":$emissionKind
+    "DIEmissionKind":$emissionKind,
+    DefaultValuedParameter<
+      "DINameTableKind", "DINameTableKind::Default">:$nameTableKind
```

Comment:
I think it still helps when parsing IR, so that even those that don't have a value will get the default value. And those with the default don't need to print, which is similar to llvm behavior. For emissionKind, llvm always prints the value, which is I think the only difference 🤔

Reply:
I thought `DINameTableKind::Default` is also omitted when printing an OptionalParameter since it is zero which is the default value. However, I may be wrong!

Reply:
Anyways making the default explicit is fine!

Reply:
ah you're right 😂 optional works too with 0 default. Changed to that.